Tests/UnitTests/Core/EventData/CMakeLists.txt

```diff
@@ -9,7 +9,9 @@ add_unittest(MultiTrajectory MultiTrajectoryTests.cpp)
 add_unittest(TransformHelpers TransformHelpersTests.cpp)
 add_unittest(CorrectedTransformFreeToBound CorrectedTransformFreeToBoundTests.cpp)
 add_unittest(Track TrackTests.cpp)
-target_sources(ActsUnitTestTrack PUBLIC TrackTestsExtra.cpp)
+if(("${ACTS_BUILD_ONLY_MATCHING_UNITTESTS}" STREQUAL "") OR ("ActsUnitTestTrack" MATCHES "${ACTS_BUILD_ONLY_MATCHING_UNITTESTS}"))
+  target_sources(ActsUnitTestTrack PUBLIC TrackTestsExtra.cpp)
```

Comment:
Maybe we can just extend the macro to accept multiple source files, and a list of targets. The we can roll these extra calls into the main macro call, and it falls within the if on its own. 

What do you think @benjaminhuth?

Reply:
Yeah, that sounds good.
But before I get my hands on it, we should decide if we want to go forward in this direction at all